rules/typescript-enums/rule.md

```diff
@@ -0,0 +1,100 @@
+---
+type: rule
+title: Do you know why to use const assertions instead of TypeScript enums?
+uri: typescript-enums
+authors:
+  - title: Harry Ross
+    url: https://ssw.com.au/people/harry-ross
+related: 
+  - use-enums-instead-of-hard-coded-strings
+created: 2024-03-19T21:39:38.906Z
+archivedreason: null
+guid: ba19be99-354d-44b2-a2da-4131cc660f18
+---
+It's super important to ensure that [magic strings are not used in your codebase](https://www.ssw.com.au/rules/use-enums-instead-of-hard-coded-strings/). Typically, we would use enums to solve this problem, but this may not be applicable when using TypeScript. You might expect TypeScript enums to function like strongly typed languages like C# and Rust but often this is not the case.

```

Comment:
Not every magic strings need `enum`, sometimes they just need a const to label them :)
e.g. `INPUT_DEBOUNCE_TIME`

I think we should reword this part to highlight when and why we should use enum in TS.

My go-to preference is to use `enum` to signify a group of **unique** values that is different than other values.
So in case like the video where Matt Pocock is trying use the `log` function is the incorrect use of enum - `enum` does not represent values, but they represent the _identity_ of the collection 😄 

In this case, TS is correct to show that it is an invalid assignment, since they are, essentially, different.
Following the video example on [6:13](https://youtu.be/jjMbPt_H3RQ?feature=shared&t=373)

```ts
// Developer has to intentionally cast it so TS understand that it is a forced assignment
log('hello world', LogLevel2.Debug as unknown as LogLevel);
```

Reply:
Done - added a sentence in the last paragraph referencing where an enum may be better choice.